src/Plugins/ComplexCore/test/RawBinaryReaderTest.cpp

```diff
@@ -0,0 +1,446 @@
+  args.insertOrAssign(RawBinaryReaderFilter::k_NumberOfComponents_Key, std::make_any<uint64>(N));
+  args.insertOrAssign(RawBinaryReaderFilter::k_Endian_Key, std::make_any<ChoicesParameter::ValueType>(Detail::Little));
+  args.insertOrAssign(RawBinaryReaderFilter::k_SkipHeaderBytes_Key, std::make_any<uint64>(skipBytes));
+  args.insertOrAssign(RawBinaryReaderFilter::k_CreatedAttributeArrayPath_Key, k_CreatedArrayPath);
+
+  return args;
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+template <typename T>
+bool createAndWriteToFile(std::string filePath, T* dataArray, usize dataSize)
+{
+  // Create the output file to dump some data into
+  FILE* f = fopen(filePath.c_str(), "wb");
+
+  // Write the data to the file
+  usize numWritten = 0;
+  while(1)
+  {
+    numWritten += fwrite(dataArray, sizeof(T), dataSize, f);
+    if(numWritten == dataSize)
+    {
+      break;
+    }
+    dataArray = dataArray + numWritten;
+  }
+
+  // Close the file
+  fclose(f);
+
+  // Return successful
+  return true;
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+// Case1: This tests when skipHeaderBytes equals 0, and checks to see if the data read is the same as the data written.
+template <typename T, usize N>
+void testCase1_Execute(complex::NumericType scalarType)
+{
+  usize dataArraySize = 10000000 * N;
+  usize skipHeaderBytes = 0;
+  DataStructure ds;
+
+  // Allocate an array, and get the dataArray from that array
+  std::shared_ptr<DataStore<T>> store = std::make_shared<DataStore<T>>(dataArraySize);
+  T* dataArray = store->data();
+
+  // Write some data into the data array
+  for(usize i = 0; i < dataArraySize; ++i)
+  {
+    dataArray[i] = static_cast<T>(i);
+  }
+
+  // Create scope guard to remove file after this test goes out of scope
+  fs::path testPath = k_TestOutput;
+  auto fileGuard = MakeScopeGuard([testPath]() noexcept { fs::remove(testPath); });
+
+  // Create the file and write to it.  If any of the information is wrong, the result will be false
+  bool result = createAndWriteToFile(k_TestOutput, dataArray, dataArraySize);
+
+  // Test to make sure that the file was created and written to successfully
+  REQUIRE(result == true);
+
+  // Create the filter, passing in the skipHeaderBytes
+  RawBinaryReaderFilter filt;
+  Arguments args = createFilterArguments(scalarType, N, skipHeaderBytes);
+
+  // Preflight, get the error condition, and check that there are no errors
+  auto preflightResult = filt.preflight(ds, args);
+  REQUIRE(preflightResult.outputActions.valid());
+
+  // Execute the filter, check that there are no errors, and compare the data
+  auto executeResult = filt.execute(ds, args);
+  REQUIRE(executeResult.result.valid());
```

Comment:
```suggestion
  COMPLEX_RESULT_REQUIRE_VALID(executeResult.result);
```
`COMPLEX_RESULT_REQUIRE_VALID` from `complex/UnitTest/UnitTestCommon.hpp` prints out additional information from `Result` on failure and should be preferred over a plain `REQUIRE` in these cases.